### DISCUSSION TITLE
workflow_dispatch yaml config doesnt work as documented

### DISCUSSION THREAD

**Discussion by rukai** [community/community#7041] [Actions] [answered]
https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#example
The documentation says I can use it like this but it doesnt work. It seems to fail to parse the yaml giving me an error with the line number of the workflow_dispatch.
```yaml
on:
  workflow_dispatch
```

Instead it works when I define an empty yaml map like:
```yaml
on:
  workflow_dispatch: {}
```

As a bonus bug, when I define like:
```yaml
on:
  workflow_dispatch: []
```
The github UI stops working!
Pressing the "Run workflow" button causes a popup to appear that is stuck infinitely loading.
![image](https://user-images.githubusercontent.com/5120858/139528049-137654e3-3d48-4875-9ecf-f581976b8da1.png)

**Comment by Slowlife01** [comment]
If it's multiple events you would have to make it like this or yea use `{}`/`null`:
```yaml
on: 
  workflow_dispatch:
  pull_request:
```
because of course that's just how YAML works

```yaml
on: 
  workflow_dispatch
```

```yaml
on: workflow_dispatch
```

otherwise those should work just fine...
and you aren't supposed to set it to array anyways

**Reply by rukai** [reply]
Yes I understand, however:
* The documentation should use this correct syntax
* Using the wrong syntax shouldn't make the UI hang. It should give me an error instead!

**Reply by Slowlife01** [reply]
- I agree with that
- It did give me an error and never shown the button. If you refresh the site it would be gone I suppose?

**Reply by Slowlife01** [reply]
Apparently I was wrong on second one, my bad I misremembered. Although it did of course give you an error on github editor, but didn't give you config error on actions tab

**Comment by ethomson** [comment, marked as answer]
Sorry I didn't see this at the time, but I can confirm that the documentation matches the product behavior now.  Thanks for the feedback!
